packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -150,6 +150,26 @@ export interface BasicCrawlerOptions<
      */
     handleRequestTimeoutSecs?: number;
 
+    /**
+     * A function to handle requests that failed less than `option.maxRequestRetries` times
+     * and would be retried by the crawler by default.
+     * Note that if the function is user-provided, it would not reclaim the request automatically.
```

Comment:
we should reclaim the request, the purpose here is to allow adjusting the request that will be retried, e.g. labeling it, changing some userData, ...

it should be kind of a hook

Reply:
E.g. you could have a pre nav hook that will make something with the request based on the label, e.g. add some cookies or whatnot. Or you could even try to adjust the payload if its PUT/POST...